File: backends/go/internal/repository/task_repository.go
package repository

import "multi-lang-backend-go/internal/domain"

type TaskRepository interface {
	Save(task *domain.Task) error
	FindByID(id string) (*domain.Task, error)
	FindByUserID(userID string) ([]*domain.Task, error)
	DeleteByID(id string) error
}

File: backends/java/src/main/java/com/example/multilang/repository/TaskRepository.java
package com.example.multilang.repository;

import com.example.multilang.domain.Task;

import java.util.List;
import java.util.Optional;

public interface TaskRepository {
    Task save(Task task);
    Optional<Task> findById(String id);
    List<Task> findByUserId(String userId);
    void deleteById(String id);
}
